### DISCUSSION TITLE
tinacms: command not found on Netlify

### DISCUSSION THREAD

**Discussion by mkz71** [tinacms/tinacms#3843] [Help] [answered]
I got this error when I deploy on Netlify.
Local environment development is fine.
Not sure what is wrong.

`$ cd exampleSite && yarn tinacms build && hugo --gc --themesDir ../..`
`yarn run v1.22.19`
`error Command "tinacms" not found.`

**Comment by kldavis4** [comment]
That indicates you need to run `yarn install` prior to executing `yarn tinacms build`

**Reply by mkz71** [reply]
I did but it is still not working so that makes me confused.

**Reply by kldavis4** [reply]
What's the error after adding the `yarn install`?

**Reply by mkz71** [reply]
Here is what I receive after run `yarn install`.

`$ yarn install`
`➤ YN0000: ┌ Resolution step`
`➤ YN0002: │ @graphiql/react@npm:0.17.1 [52a78] doesn't provide @codemirror/language (p11bf9), requested by codemirror-graphql`
`➤ YN0002: │ @graphiql/react@npm:0.17.1 [52a78] doesn't provide react-is (p96655), requested by @reach/menu-button`
`➤ YN0002: │ @tinacms/cli@npm:1.5.6 doesn't provide graphql-tag (p98de0), requested by @graphql-codegen/typescript-generic-sdk`
`➤ YN0002: │ @tinacms/cli@npm:1.5.6 doesn't provide postcss (pafccf), requested by postcss-nested`
`➤ YN0002: │ @tinacms/cli@npm:1.5.6 doesn't provide react (pe84dd), requested by @tinacms/schema-tools`
`➤ YN0002: │ @tinacms/graphql@npm:1.4.7 doesn't provide react (p87f78), requested by @tinacms/schema-tools`
`➤ YN0002: │ @tinacms/mdx@npm:1.3.8 doesn't provide react (paa226), requested by @tinacms/schema-tools`
`➤ YN0002: │ @tinacms/mdx@npm:1.3.8 doesn't provide yup (p79f22), requested by @tinacms/schema-tools`
`➤ YN0002: │ @udecode/plate-serializer-csv@npm:14.4.2 [7b40b] doesn't provide slate-history (pf52bf), requested by @udecode/plate-core`
`➤ YN0002: │ @udecode/plate-serializer-csv@npm:14.4.2 [7b40b] doesn't provide slate-history (p43418), requested by @udecode/plate-table`
`➤ YN0002: │ @udecode/plate-serializer-html@npm:14.4.2 [7b40b] doesn't provide slate-history (p95bbc), requested by @udecode/plate-core`
`➤ YN0002: │ @udecode/zustood@npm:1.1.3 [13868] doesn't provide react (p67626), requested by react-tracked`
`➤ YN0002: │ @udecode/zustood@npm:1.1.3 [13868] doesn't provide scheduler (p9dc88), requested by react-tracked`
`➤ YN0002: │ examplesite@workspace:. doesn't provide @babel/core (p2dd2e), requested by babel-preset-current-node-syntax`
`➤ YN0002: │ examplesite@workspace:. doesn't provide @babel/core (p2fb16), requested by babel-jest`
`➤ YN0002: │ examplesite@workspace:. doesn't provide @babel/core (pf005d), requested by babel-preset-fbjs`
`➤ YN0002: │ examplesite@workspace:. doesn't provide @babel/core (pa7b63), requested by babel-preset-jest`
`➤ YN0002: │ examplesite@workspace:. doesn't provide @codemirror/language (p5f7c7), requested by codemirror-graphql`
`➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code`
`➤ YN0000: └ Completed in 0s 336ms`
`➤ YN0000: ┌ Fetch step`
`➤ YN0000: └ Completed in 0s 438ms`
`➤ YN0000: ┌ Link step`
`➤ YN0000: └ Completed in 0s 549ms`
`➤ YN0000: Done with warnings in 1s 411ms`

**Reply by kldavis4** [reply]
That looks correct and I don't see an error. After running install, you should be able to execute tinacms. The full command would something like: `cd exampleSite && yarn install && yarn tinacms build && hugo --gc --themesDir ../..`

**Reply by mkz71** [reply]
I got this error after running: `cd exampleSite && yarn install && yarn tinacms build && hugo --gc --themesDir ../..`

 `$ /opt/build/repo/exampleSite/node_modules/.bin/tinacms build`
`Starting Tina build`
`ERROR: Branch 'master' is not on Tina Cloud. Please make sure that branch 'master' exists in your repository and that you have pushed your all changes to the remote. View all all branches and there current status here: `

And when I look at this project in app.tina.io under Configuration, Branch Status shows No branches found.

**Reply by kldavis4** [reply]
That looks like progress :)

The issue now is that tina cloud doesn't know where the tina/tina-lock.json is located in your repo. It only checks the default branch ('master' in this case) when you create a new project. First verify that you have that pushed up to your repository in github. Then, if it is in a sub-folder and not the root of your repository, then you have to specify the path to the subfolder. It might be `exampleSite` in this case. See https://tina.io/docs/tina-cloud/dashboard/projects/#path-to-tina for reference.

**Reply by mkz71** [reply, marked as answer]
After adding `yarn install` to my previous command `cd exampleSite && yarn tinacms build && hugo --gc --themesDir ../..`  in netify.toml file.

Now the netify.toml file looke like this:
`cd exampleSite && yarn install && yarn tinacms build && hugo --gc --themesDir ../..`
And I add "exampleSite" to Path to Tina Config in the Advanced Settings.

Now it is built successfuly!! :)

Thank you so much!!!

**Reply by jamespohalloran** [reply]
I'm kind of surprised Netlify isn't running install automatically. From this doc, it looks like it should be: https://docs.netlify.com/configure-builds/manage-dependencies/
but maybe that varies based on the build image?

**Reply by mkz71** [reply]
This might not be related to tinacms directly.
I have another repo and I setup the same as this one but after I deploy it on Netlify.
I got this:

`$ cd exampleSite && yarn install && yarn tinacms build && hugo --gc --themesDir ../..`
`bash: yarn: command not found`

It is weird.

**Reply by jamespohalloran** [reply]
What is the Netlify build runtime used for both projects? Do you have a `yarn.lock` file pushed up for this repo?

**Reply by mkz71** [reply]
Yes, both have @netlify/build 29.9.2 and `yarn.lock` in those repo.
I look at the Deploy log and I see that the one build successfully is using Ruby version 2.7.2 but the second is using Ruby version 2.7.1.
I am not sure this is the case.

**Reply by mkz71** [reply]
I don't know why the second repo is using Node.js v12.18.0 so I add this NODE_VERSION to 16.20.0 in Netlify Environment setup.
The same as the first one that I build successfully.
Now the second repo is built successfully now.
How can I update the Netlify for this repo without adding this environment value?
